contracts/SeriesIndexer.sol

```diff
@@ -0,0 +1,792 @@
+
+    /**
+     * @dev Revokes owner rights for the calling artist
+     */
+    function revokeOwnerRightsForArtist() external {
+        uint256 artistID = addressToArtistID[msg.sender];
+        if (artistID == 0) {
+            revert NotAnArtistError(msg.sender);
+        }
+        if (_isArtistOwnerRightRevoked(artistID)) {
+            return;
+        }
+        _setArtistOwnerRightRevokedBool(artistID, true);
+    }
+
+    /**
+     * @dev Approves owner rights for the calling artist
+     */
+    function approveOwnerRightsForArtist() external {
+        uint256 artistID = addressToArtistID[msg.sender];
+        if (artistID == 0) {
+            revert NotAnArtistError(msg.sender);
+        }
+        if (!_isArtistOwnerRightRevoked(artistID)) {
+            return;
+        }
+        _setArtistOwnerRightRevokedBool(artistID, false);
+    }
+
+    /**
+     * @dev Updates an artist's address
+     */
+    function updateArtistAddress(uint256 artistID, address newAddress) external {
+        if (newAddress == address(0)) {
+            revert InvalidNewAddressError(newAddress);
+        }
+        if (addressToArtistID[newAddress] != 0) {
+            revert AddressAlreadyAssignedError(newAddress);
+        }
+        if (artists[artistID].artistAddress == address(0)) {
+            revert InvalidArtistIDError(artistID);
+        }
+
+        uint256 callerID = addressToArtistID[msg.sender];
+        bool isCallerArtist = (callerID == artistID && callerID != 0);
+        bool isOwnerWithRights = (msg.sender == owner() && !_isArtistOwnerRightRevoked(artistID));
+
+        if (!isCallerArtist && !isOwnerWithRights) {
+            revert NotAuthorizedError(msg.sender);
+        }
+
+        address oldAddress = artists[artistID].artistAddress;
+        addressToArtistID[oldAddress] = 0;
+        artists[artistID].artistAddress = newAddress;
+        addressToArtistID[newAddress] = artistID;
+
+        emit ArtistAddressUpdated(artistID, oldAddress, newAddress);
+    }
+
+    // ============ Public/External View Functions ============
+
+    /**
+     * @notice Returns `true` if at least one artist in the series
+     *         has not revoked the owner's right to modify the series.
+     */
+    function ownerCanModifySeries(uint256 seriesID) public view returns (bool) {
+        uint256[] memory ids = seriesDetails[seriesID].artistIDs;
+        if (ids.length == 0) {
+            // If no artists, owner can modify the series
+            return true;
+        }
+        for (uint256 i = 0; i < ids.length; i++) {
+            // If any artist has not revoked, return true
+            if (!_isArtistOwnerRightRevoked(ids[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Checks if the given `artistAddr` has revoked owner rights
+     */
+    function ownerRightsRevoked(address artistAddr) external view returns (bool) {
```

Comment:
Could be `hasArtistRevokedOwnerRights` for more clear.
Also shouldn't use abbreviation for parameter.